### PR TITLE
Backend to show whether pinned messages are read

### PR DESCRIPTION
**Pull request by megrogan**

### CODE REVIEW

**Comment by hpeebles** [comment]
AFAICT this can't handle a few cases which aren't all that uncommon.
1 - if a user pins a message, then unpins it, everyone will see that there is a new pinned message when there isn't.
2 - if a user reads a pinned message, but then that message is unpinned and pinned again it'll appear as unread again when it isn't.

I think we should be handling these cases.

**Review by hpeebles — DISMISSED** [review]

**Review by hpeebles — APPROVED** [review]

**Review by hpeebles — DISMISSED** [review]

File: backend/canisters/user/impl/src/updates/mark_read_v2.rs [hunk 1]
```diff
@@ -29,6 +30,11 @@ fn mark_read_impl(args: Args, runtime_state: &mut RuntimeState) -> Response {
                     .threads_read
                     .insert(thread.root_message_index, thread.read_up_to, now);
             }
+
+            group_chat.date_read_pinned = Timestamped::new(
```
Comment (hpeebles) [inline backend/canisters/user/impl/src/updates/mark_read_v2.rs:34]: This is always bumping the timestamp, we don't want that.

File: backend/canisters/user/impl/src/updates/mark_read_v2.rs [hunk 2]
```diff
@@ -29,6 +30,13 @@ fn mark_read_impl(args: Args, runtime_state: &mut RuntimeState) -> Response {
                     .threads_read
                     .insert(thread.root_message_index, thread.read_up_to, now);
             }
+
+            if chat_messages_read.date_read_pinned.is_some() {
```
Comment (hpeebles) [inline backend/canisters/user/impl/src/updates/mark_read_v2.rs:34]: Maybe this?

`if chat_messages_read.date_read_pinned > group_chat.date_Read_pinned.value {
...
}`

Comment (megrogan) [reply backend/canisters/user/impl/src/updates/mark_read_v2.rs:34]: Yes I was just reviewing this

**Pull request merged**